### ISSUE TITLE
Translucency and images with Alpha channel

### ISSUE THREAD

**Issue by tehsphinx** [fyne-io/fyne#1977]
### Describe the bug:

When applying Translucency to an image with an alpha channel, the otherwise transparent part of the image becomes visible. Below example uses one of the default icons to show the effect.

### To Reproduce:
Steps to reproduce the behaviour:
1. Run the go code below. `go run .`

### Screenshots:

![image](https://user-images.githubusercontent.com/6624571/108523916-c2aed800-72ce-11eb-8d59-7a0c599e73df.png)

Note: The background of the icon should not be darkened.

### Example code:

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/canvas"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/theme"
)

func main() {
	a := app.New()

	w := a.NewWindow("Test App")
	image := canvas.NewImageFromResource(theme.ConfirmIcon())
	image.Translucency = 0.5
	w.SetContent(container.NewPadded(image))
	w.Resize(fyne.Size{
		Width:  100,
		Height: 100,
	})

	w.ShowAndRun()
}
```

### Device (please complete the following information):
 - MacOS
 - Big Sur 11.2.1
 - go version go1.16 darwin/amd64
 - fyne/v2 2.0.0

**Comment by tehsphinx** [comment]
Might not be that important...

**Comment by AlbinoGeek** [comment]
Thank you for bringing this to attention, I wasn't sure if I'd simply done something wrong (could not reproduce cleanly.)

**Comment by andydotxyz** [comment]
Yes, thanks. We currently support an alpha channel or a translucency value but not both.
This should be fixed.

**Comment by junmaqiang** [comment]
me too  I have this ask

**Comment by junmaqiang** [comment]
# this is bug?
`
img := canvas.NewImageFromResource(theme.VisibilityOffIcon())
	img.FillMode = canvas.ImageFillContain
	img.SetMinSize(fyne.NewSize(237, 237))
	img.Translucency = 0.5
`

<img width="550" alt="截屏2023-09-10 14 05 22" src="https://github.com/fyne-io/fyne/assets/1819457/79594d22-4a7e-47d0-8d1a-89ea9dcc5433">

**Comment by andydotxyz** [comment]
Yes

**Comment by hkparker** [comment]
Running into this now as well.  How difficult would it be to address this?

**Comment by andydotxyz** [comment]
I think the error is in the GL shader - so not an easy fix as far as I can estimate.
However there is a possibility that it relates to colour models in the SVG rasteriser

**Comment by hkparker** [comment]
Ah, in this case I was using a PNG, so it wouldn't be in any SVG specific code I expect

**Comment by andydotxyz** [comment]
Thanks for the prod. I finally found the solution 🎉

**Comment by andydotxyz** [comment]
This is landed on `develop` and in the `v2.5.3-rc3` that we are testing
